Templates/GraphicsGem/Template/Code/Source/Components/${Name}ComponentController.h

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright (c) Contributors to the Open 3D Engine Project.
+ * For complete copyright and license terms please see the LICENSE at the root of this distribution.
+ *
+ * SPDX-License-Identifier: Apache-2.0 OR MIT
+ *
+ */
+
+#pragma once
+
+#include <AzCore/Asset/AssetCommon.h>
+#include <AzCore/Component/Component.h>
+#include <AzCore/Component/TransformBus.h>
+
+#include <${Name}/${Name}FeatureProcessorInterface.h>
+
+namespace ${Name}
+{
+    class ${Name}ComponentConfig final
+        : public AZ::ComponentConfig
+    {
+    public:
+        AZ_RTTI(${Name}ComponentConfig, "{${Random_Uuid}}", ComponentConfig);
+        AZ_CLASS_ALLOCATOR(${Name}ComponentConfig, AZ::SystemAllocator);
+        static void Reflect(AZ::ReflectContext* context);
+
+        ${Name}ComponentConfig() = default;
+
+        AZ::u64 m_entityId{ AZ::EntityId::InvalidEntityId };
+    };
```

Comment:
nit I would consider moving the config to its own h/cpp files but it's not critical.